### ISSUE TITLE
Style/PercentLiteralDelimiters doesn't detect `%{}`

### ISSUE THREAD

**Issue by vfonic** [rubocop-hq/rubocop#7266]
I was just running rubocop constantly, when it, out of a sudden, fixed one `%{ }` multi-line string delimeter to `%( )`. 🤷‍♂ 

Just next to it there are tons of not-detected `%{ }` strings.

After trying out, I've managed to narrow it down to this:

```rb
# frozen_string_literal: true

a = %{()}
puts a
```

If there are `(` and `)` (actually only `(` would do) in the string, the cop won't raise error.

Update:

```yaml
Style/PercentLiteralDelimiters:
  PreferredDelimiters:
    default: '{}'
```
With this setup, it fails to find `%({)` as error.

## RuboCop version

```
$ rubocop -V
0.74.0 (using Parser 2.6.3.0, running on ruby 2.6.3 x86_64-darwin18)
```

**Comment by tejasbubane** [comment]
@vfonic This is expected behaviour. See https://github.com/rubocop-hq/rubocop/blob/69bce3fd009a4c2723f3f52e10c3fc54f826df0e/spec/rubocop/cop/style/percent_literal_delimiters_spec.rb#L77-L82 Although I am not sure why we need it.

**Comment by vfonic** [comment]
@tejasbubane I'm not sure the test you linked is what I am reporting in this issue.

```rb
  it 'does not register an offense for other delimiters ' \ 
        'when containing preferred delimiter characters' do 
       expect_no_offenses(<<~RUBY) 
         %q([string]) 
       RUBY 
     end 
```

The `%q()` is the default/preferred delimiter. I'm reporting that, for this case: `%()` (also preferred delimiter, but without `q`), it doesn't report anything, for this case: `%{(}`.

So this code wouldn't report any errors:

```rb
asd = %{(}
dsa = %({)
```

...while, obviously both can't be default/preferred delimiters.

**Comment by buehmann** [comment]
@vfonic The test uses `[` as the preferred delimiter: https://github.com/rubocop-hq/rubocop/blob/69bce3fd009a4c2723f3f52e10c3fc54f826df0e/spec/rubocop/cop/style/percent_literal_delimiters_spec.rb#L7

So this is equivalent to your case: https://github.com/rubocop-hq/rubocop/blob/69bce3fd009a4c2723f3f52e10c3fc54f826df0e/spec/rubocop/cop/style/percent_literal_delimiters_spec.rb#L49-L54

This has been in the cop since its introduction. The PR has some discussion about it: https://github.com/rubocop-hq/rubocop/pull/834#issuecomment-36253133 ff.

**Comment by vfonic** [comment]
Ah I see. So, to me, this seems like an issue.

Why would this test be written? I can't wrap my head around this test. 🤔 

/cc @bbatsov @hannestyden

**Comment by hannestyden** [comment]
@vfonic I don't remember my exact reasoning at the time, but I would assume the test was added to ensure that the cop is lenient in cases where e.g. `%([])` would be thought to be easier (for humans) to parse than `%[[]]`, even if `[` and `]` were the preferred delimiters.

I hope that helps. In any case I don't have strong opinions about the behavior of the `PercentLiteralDelimiters` cop, so feel free to change it any way you see fit.

**Comment by vfonic** [comment]
That's actually the reason why I opened this issue. 😁
I wanted to write my graphql queries like this:

```
%{
{
 events {
 id
}
}
}
```

Sorry for formatting, I'm on my phone.

This didn't show any errors when I had () inside.

But then, some of my queries (mutations actually) had () so half of my
queries were `%{` and half were `%(`, which brought me here. 🤷‍♂️

I think better solution would be to let the user choose to disable that
rubocop check with the inline comment, if he wants to.

**Comment by Drenmi** [comment]
It could potentially be that the cop is trying to predict the user (not always a bad thing, but it's always a fine line) a bit too much? 🙂
